src/xrpld/app/tx/detail/InvariantCheck.cpp

```diff
@@ -1493,6 +1498,15 @@ ValidMPTIssuance::finalize(
 {
     if (result == tesSUCCESS)
     {
+        if (mptCreatedByIssuer_)
+        {
+            JLOG(j.fatal())
+                << "Invariant failed: MPT authorize submitted by issuer "
+                   "succeeded but created mptokens";
+            return false;
```

Comment:
i believe this needs to be amendment gated

Reply:
> i believe this needs to be amendment gated

Agreed. I would have the gate check for either `SingleAssetVault` or `LendingProtocol` to be enabled.